jwst/extract_2d/grisms.py

```diff
@@ -576,12 +580,44 @@ def compute_wavelength_array(slit):
     wavelength : numpy.array
         The wavelength array
     """
-    transform = slit.meta.wcs.forward_transform
-    x, y = grid_from_bounding_box(slit.meta.wcs.bounding_box)
-    wavelength = transform(x, y)[2]
+    wcs = slit.meta.wcs
+    full_transform = slit.meta.wcs.forward_transform
+    x, y = grid_from_bounding_box(wcs.bounding_box)
+    wavelength = full_transform(x, y)[2]
     return wavelength
 
 
+def compute_tso_offset_center(input_model: ImageModel, distortion: CompoundModel) -> tuple[float, float]:
+    """
+    In the case that an Offset Special Requirement is requested in the APT,
+    the source is no longer at the pointing reference.
+    The dither.x_offset and dither.y_offset values encode the offset,
+    and it needs to be translated from Ideal to detector coordinates.
+
+    Parameters
+    ----------
+    input_model : ImageModel
+        The input data model
+    distortion : DistortionModel
+        The distortion model
+
+    Returns
+    -------
+    xc, yc : tuple
+        The x and y center of the image in direct image coordinates
+    """
+
+    idltov23 = IdealToV2V3(input_model.meta.wcsinfo.v3yangle,
+                input_model.meta.wcsinfo.v2_ref,
+                input_model.meta.wcsinfo.v3_ref,
+                input_model.meta.wcsinfo.vparity)
+    v2_offset, v3_offset = idltov23(input_model.meta.dither.x_offset, input_model.meta.dither.y_offset)
+    wavelength = float(input_model.meta.instrument.filter[1:4])/100 
```

Comment:
Hmmm, `filter[1:4]` should work OK for any data from NIRCam, for which all the filters are less than 10 microns and hence only have 3 digits in their name. But some MIRI filters that are beyond 10 microns have 4 digits. Now the question is whether this would ever be used for any MIRI data. I guess right now the routine is only ever being called for the NIRCam TSGrism mode, so I guess it's safe. We really need a header keyword that gives the central/pivot wavelength of the filter in use (need that in wfss_contam too).

Reply:
On second thought, does the wavelength value used as input to the `distortion` transform even make a difference to the answer you get?  If not, we could potentially just hardwire the value to something like 1.

Reply:
true, it does not matter between v2v3 and direct image coordinates